python_modules/libraries/dagster-aws/dagster_aws_tests/secretsmanager_tests/test_secrets.py

```diff
@@ -98,21 +101,62 @@ def test_secretmanager_secrets_resource(mock_secretsmanager_resource):
         assert os.getenv("baz_secret") is None
         assert os.getenv("asdf_secret") is None
 
+    # Test adding secrets to the environment
+    # Ensure env vars are overridden
+    with environ({"foo_secret": "prior_foo_value"}):
+        assert os.getenv("foo_secret") == "prior_foo_value"
+
+        with secretsmanager_secrets_resource(
+            build_init_resource_context(
+                config={
+                    "secrets_tag": "dagster",
+                    "secrets": [asdf_secret["ARN"]],
+                    "add_to_environment": True,
+                }
+            )
+        ) as secret_map:
+            assert secret_map == {
+                "foo_secret": "foo_value",
+                "baz_secret": "baz_value",
+                "asdf_secret": "asdf_value",
+            }
+
+            assert os.getenv("foo_secret") == "foo_value"
+            assert os.getenv("baz_secret") == "baz_value"
+            assert os.getenv("asdf_secret") == "asdf_value"
+
+    # Test JSON secret
+    json_secret_obj = {"key1": "value1", "key2": {"nest1": "value2", "nest2": "value3"}}
+    json_secret = mock_secretsmanager_resource.create_secret(
+        Name="json_secret",
+        SecretString=json.dumps(json_secret_obj),
+    )
     with secretsmanager_secrets_resource(
         build_init_resource_context(
             config={
-                "secrets_tag": "dagster",
-                "secrets": [asdf_secret["ARN"]],
+                "secrets": [json_secret["ARN"]],
+                "add_to_environment": True,
+            }
+        )
+    ) as secret_map:
+        assert json.loads(secret_map["json_secret"]) == json_secret_obj
+
+        assert json.loads(os.getenv("json_secret")) == json_secret_obj
+
+    # Binary secrets have a None value
+    binary_secret = mock_secretsmanager_resource.create_secret(
+        Name="binary_secret", SecretBinary=b"binary_value"
+    )
+    with secretsmanager_secrets_resource(
+        build_init_resource_context(
+            config={
+                "secrets": [binary_secret["ARN"]],
                 "add_to_environment": True,
             }
         )
     ) as secret_map:
         assert secret_map == {
-            "foo_secret": "foo_value",
-            "baz_secret": "baz_value",
-            "asdf_secret": "asdf_value",
+            "binary_secret": None,
```

Comment:
Cool - not entirely sure this is the behavior we'll want long term, but at least now we have a breadcrumb to show how we currently expect it to work.